cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala

```diff
@@ -0,0 +1,396 @@
+
+  def setEndTimeCol(v: String): this.type = setVectorParam(endTime, v)
+
+  def getEndTime: String = getScalarParam(endTime)
+
+  def getEndTimeCol: String = getVectorParam(endTime)
+}
+
+trait MADBase extends HasAsyncReply with HasMADSource with HasMADStartTime with HasMADEndTime
+  with HasCognitiveServiceInput with HasInternalJsonOutputParser with HasSetLocation with Wrappable
+  with HTTPParams with HasOutputCol with HasURL with ComplexParamsWritable
+  with HasSubscriptionKey with HasErrorCol with BasicLogging {
+
+  setDefault(
+    outputCol -> (this.uid + "_output"),
+    errorCol -> (this.uid + "_error"))
+
+  protected def queryForResult(key: Option[String],
+                               client: CloseableHttpClient,
+                               location: URI): Option[HTTPResponseData] = {
+    val get = new HttpGet()
+    get.setURI(location)
+    key.foreach(get.setHeader("Ocp-Apim-Subscription-Key", _))
+    get.setHeader("User-Agent", s"synapseml/${BuildInfo.version}${HeaderValues.PlatformInfo}")
+    val resp = convertAndClose(sendWithRetries(client, get, getBackoffs))
+    get.releaseConnection()
+    val fields = IOUtils.toString(resp.entity.get.content, "UTF-8").parseJson.asJsObject.fields
+    var status = "none"
+    if (fields.keySet.contains("modelInfo")) {
+      status = fields("modelInfo").asInstanceOf[JsObject].fields
+        .get("status").map(_.convertTo[String]).get.toLowerCase()
+    } else if (fields.keySet.contains("summary")) {
+      status = fields("summary").asInstanceOf[JsObject]
+        .fields.get("status").map(_.convertTo[String]).get.toLowerCase()
+    }
+    status match {
+      case "ready" | "failed" => Some(resp)
+      case "created" | "running" => None
+      case s => throw new RuntimeException(s"Received unknown status code: $s")
+    }
+  }
+
+  protected def handlingFunc(client: CloseableHttpClient,
+                             request: HTTPRequestData): HTTPResponseData = {
+    val response = HandlingUtils.advanced(getBackoffs: _*)(client, request)
+    if (response.statusLine.statusCode == 201) {
+      val location = new URI(response.headers.filter(_.name == "Location").head.value)
+      val maxTries = getMaxPollingRetries
+      val key = request.headers.find(_.name == "Ocp-Apim-Subscription-Key").map(_.value)
+      val it = (0 to maxTries).toIterator.flatMap { _ =>
+        queryForResult(key, client, location).orElse({
+          blocking {
+            Thread.sleep(getPollingDelay.toLong)
+          }
+          None
+        })
+      }
+      if (it.hasNext) {
+        it.next()
+      } else {
+        throw new TimeoutException(
+          s"Querying for results did not complete within $maxTries tries")
+      }
+    } else {
+      response
+    }
+  }
+}
+
+object MultivariateAnomalyModel extends ComplexParamsReadable[MultivariateAnomalyModel] with Serializable
+
+class MultivariateAnomalyModel(override val uid: String) extends Estimator[DetectMultivariateAnomaly]
```

Comment:
Maybe we can rename this to MultivariateAnomalyEstimator so as not to confuse with the model that it spits out

Reply:
Sure. Done.